### DISCUSSION TITLE
Kafka TLS: Thoughts on supporting PKCS #8 PEM private key?

### DISCUSSION THREAD

**Discussion by dtanner** [redpanda-data/connect#1519] [Q&A] [answered]
Reading [the docs](https://www.benthos.dev/docs/components/outputs/kafka#tlsclient_certspassword) for `tls.client_certs[].password`, it looks like there's a single supported encryption format of PKCS #1.  Is that correct?  If so, any plans to support other formats?  

Background:
I got a `tls: failed to parse private key` error when trying to use a password-protected, PEM-formatted PKCS8 private key as my `key_file` for a kafka output. The same file works with other tools like `kcat` and the java kafka client.  The java client doesn't support PKCS #1 😢 but does support PKCS #8 (only with passwords though!), and I'm trying to find a single compatible format for all my kafka tooling.  
I resolved the issue by using an unencrypted PKCS #8 PEM file, which does work in Benthos.  So hopefully I'm missing something, but if not I'm curious if there's anything on the roadmap for encryption support.

**Comment by mihaitodor** [comment, marked as answer]
Ugh, I managed to let this one slip through the cracks... Thanks for bearing with us and happy to help! I created issue #1565 to address this. Let's work out the implementation plan there.

**Reply by mihaitodor** [reply]
#1569 should address this and it just got merged. Would be great to confirm that it does indeed work. I left some comments [here](https://github.com/benthosdev/benthos/issues/1565#issuecomment-1312937042) on the limitations of this implementation.

**Reply by dtanner** [reply]
Awesome thanks! i should be able to test this on monday or tuesday

**Reply by dtanner** [reply]
Confirmed this works now - thanks @mihaitodor !

**Reply by mihaitodor** [reply]
Awesome, thanks for testing @dtanner! There will be a new release soon and it will include this enhancement.
